### PR TITLE
refactor: Use workspace dependencies, move crates into common folder

### PR DESCRIPTION
**Pull request by Techassi**
Part of https://github.com/stackabletech/issues/issues/531, fixes https://github.com/stackabletech/operator-rs/issues/737

### CODE REVIEW

File: crates/stackable-operator-derive/Cargo.toml [hunk 1]
```diff
@@ -1,7 +1,7 @@
 [package]
 name = "stackable-operator-derive"
 description = "Derive macros for the Stackable Operator Framework"
-version.workspace = true
+version = "0.1.0"
```
Comment (NickLarsenNZ) [inline crates/stackable-operator-derive/Cargo.toml:4]: Why do we hard code the in each crate?
Do we want to version everything independently (wondering what the burden is)?

IMO, we should re-export the crates and use feature flags so we can simply add one dependency to the operators and toggle the features that way. Not a strong opinion though.

**Review by NickLarsenNZ — APPROVED** [review]
LGTM, left a comment (non-blocker)

File: Cargo.toml [hunk 2]
```diff
@@ -1,29 +1,21 @@
+[workspace]
+members = ["crates/*"]
```
[the remark this answers is not in the document]
Comment (Techassi) [reply Cargo.toml:2]: It is very nice indeed :)

File: crates/stackable-operator-derive/Cargo.toml (hunk 1 above)
Comment (Techassi) [reply crates/stackable-operator-derive/Cargo.toml:4]: > Do we want to version everything independently (wondering what the burden is)?

In my opinion, we should do that. Always greater flexibility. We might need some tooling in the future.

> IMO, we should re-export the crates and use feature flags so we can simply add one dependency to the operators and toggle the features that way. Not a strong opinion though.

That's part of my long-term vision. I'm working towards that goal step-by-step and we eventually will use selected re-exports for various stuff.

**Pull request merged**